New resource: `azurerm_eventgrid_partner_namespace` (#30266)
hashicorp/terraform-provider-azurerm: website/docs/r/eventgrid_partner_namespace.html.markdown

---
subcategory: "Messaging"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_eventgrid_partner_namespace"
description: |-

>   Manages an EventGrid Partner Namespace.

Reviewer: Same comment as on the data source

```suggestion
  Manages an Event Grid Partner Namespace.
```

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_eventgrid_partner_registration" "example" {
  name                = "example-partner-registration"
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_eventgrid_partner_namespace" "example" {
  name                    = "example-partner-namespace"
  location                = azurerm_resource_group.example.location
  resource_group_name     = azurerm_resource_group.example.name
  partner_registration_id = azurerm_eventgrid_partner_registration.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this EventGrid Partner Namespace. Changing this forces a new EventGrid Partner Namespace to be created.

* `location` - (Required) Specifies the Azure Region where the EventGrid Partner Namespace exists. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the Resource Group where the EventGrid Partner Namespace should exist. Changing this forces a new EventGrid Partner Namespace to be created.

> * `partner_registration_id` - (Required) The fully qualified ARM ID of the EventGrid Partner Registration that this namespace is associated with. Changing this forces a new EventGrid Partner Namespace to be created.

Reviewer: ```suggestion
* `partner_registration_id` - (Required) The resource Id of the Event Grid Partner Registration that this namespace is associated with. Changing this forces a new EventGrid Partner Namespace to be created.
```

---

* `inbound_ip_rule` - (Optional) One or more `inbound_ip_rule` blocks as defined below.

> * `local_auth_enabled` - (Optional) Whether local authentication methods is enabled for the EventGrid Partner Namespace. Defaults to `true`.

Reviewer: ```suggestion
* `local_auth_enabled` - (Optional) Whether local authentication methods are enabled for the Event Grid Partner Namespace. Defaults to `true`.
```

* `partner_topic_routing_mode` - (Optional) The partner topic routing mode. Possible values are `ChannelNameHeader` and `SourceEventAttribute`. Defaults to `ChannelNameHeader`. Changing this forces a new EventGrid Partner Namespace to be created.

* `public_network_access_enabled` - (Optional) Whether or not public network access is allowed for this server. Defaults to `true`.

* `tags` - (Optional) A mapping of tags which should be assigned to the EventGrid Partner Namespace.

---

An `inbound_ip_rule` block supports the following:

* `ip_mask` - (Required) The IP mask (CIDR) to match on.

> * `action` - (Optional) The action to take when the rule is matched. Possible values are `Allow`. Defaults to `Allow`.

Reviewer: Since there's currently only one accepted value for this could you update this to:

```suggestion
* `action` - (Optional) The action to take when the rule is matched. The only possible value is `Allow`. Defaults to `Allow`.
```
